www/source/blog/2017-11-30-Windows-Docker-Export.html.md

```diff
@@ -0,0 +1,181 @@
+
+## Export the ASP.NET Core Application
+
+```console
+hab pkg export docker core/habitat-aspnet-sample
+```
+
+Now a `docker image ls` should include something like this:
+
+```console
+REPOSITORY                                      TAG                     IMAGE ID
+CREATED             SIZE
+core/mysql                                      5.7.17                  aa93cd96fd21
+2 hours ago         12.5GB
+core/mysql                                      5.7.17-20170315123622   aa93cd96fd21
+2 hours ago         12.5GB
+core/mysql                                      latest                  aa93cd96fd21
+2 hours ago         12.5GB
+core/habitat-aspnet-sample                      0.2.0                   1a1816ecb4f7
+2 hours ago         10.7GB
+core/habitat-aspnet-sample                      0.2.0-20170703165439    1a1816ecb4f7
+2 hours ago         10.7GB
+core/habitat-aspnet-sample                      latest                  1a1816ecb4f7
+2 hours ago         10.7GB
+```
+
+## Start MySQL
+
+```console
+docker run -p 3306:3306 -it core/mysql
+```
+
+This should produce something like:
+
+```console
+hab-sup(MR): Supervisor Member-ID 69c41a05c5ba44109b7e0f9c00b93ad1
+hab-sup(MR): Starting core/mysql
+hab-sup(MR): Starting gossip-listener on 0.0.0.0:9638
+hab-sup(MR): Starting http-gateway on 0.0.0.0:9631
+mysql.default(HK): init, compiled to C:/hab/svc\mysql\hooks\init
+mysql.default(HK): run, compiled to C:/hab/svc\mysql\hooks\run
+mysql.default(HK): Hooks compiled
+mysql.default(SR): Hooks recompiled
+default(CF): Updated init.sql 8db8e9979876f6fd29559ec0e4ab35c9964dc8cb6001e450e66cd2d96c7e4a60
+default(CF): Updated my.cnf 0142697a2d8284d415b05f6bca27607a0911b92c3fe3048889fdd640e6ca3578mysql.default(SR): Configuration recompiled
+mysql.default(SR): Initializing
+mysql.default hook[init]:(HK):
+mysql.default hook[init]:(HK):
+mysql.default hook[init]:(HK):     Directory: C:\hab\svc\mysql\var
+mysql.default hook[init]:(HK):
+mysql.default hook[init]:(HK):
+mysql.default hook[init]:(HK): Mode                LastWriteTime         Length Name
+
+mysql.default hook[init]:(HK): ----                -------------         ------ ----
+
+mysql.default hook[init]:(HK): d-----       11/29/2017   1:46 PM                logs
+
+mysql.default hook[init]:(HK):
+mysql.default hook[init]:(HK):
+mysql.default(SV): Starting service as user=containeradministrator, group=
+```
+
+## Populate the Database with the Initial Schema
+
+We now have an empty MySQL database running. Before our application can succesfully interact with it, we need to migrate our schema to the running instance. We will use the [EF Core .NET Command-line Tools](https://docs.microsoft.com/en-us/ef/core/miscellaneous/cli/dotnet) to perform the migration. The easiest way to do this is to run the migration locally on your host and target the running MySQL container.
```

Comment:
What's the reason for doing the migration like this, as opposed to doing something in the `init` hook?

Reply:
The entity-framework migrations need the source code meta files to run (a bummer and something they are tracking). One thing I did for for linux (which also runs .net core) is created a separate migrator package. I might do that for windows too but thought this was good enough for now.